### PR TITLE
feat(frontend): auth Integration with frontend added

### PR DESCRIPTION
**Pull request by Deepanshu0703**
## Related Issue

- I have integrated yet made frontend(user auth ) with backend

Closes: #108 

### Describe the changes you've made

I have integrated the authentication of user from the made REST api with the frontend also solved some errors already there in the code.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Enhancement in the existing features
- [x] Other

## Mention any unusual behaviour of your code (Write NA if not)

NA

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly whereever it was hard to understand.
- [x] My changes generate no new warnings.
- [x] I have verified/tested my code by running it locally.

## Files Added/Modified

SignUp.js and its css file

## Screenshots / Videos / Gif

https://github.com/jain-rishabh-21/DoctorVerse/assets/114489502/5fd9e8e0-79b5-44c7-b035-b29a0678574b

### CODE REVIEW

**Comment by jain-rishabh-21** [comment]
@Deepanshu0703 Please change the PR title to match the lint rules, you can see the merged PRs for ref.

File: client/src/App.js [hunk 1]
```diff
@@ -14,6 +12,8 @@ function App() {
          <Route  exact path={routes[0].path} Component={routes[0].element}/>
          <Route path={routes[1].path} Component={routes[1].element}/>
          <Route path={routes[2].path} Component={routes[2].element}/>
+         <Route path={routes[3].path} Component={routes[3].element}/>
+          <Route path={routes[4].path} Component={routes[4].element}/>
```
Comment (jain-rishabh-21) [inline client/src/App.js:16]: indentaion

File: client/src/pages/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    <div className='rectanglePurpleRight'></div>
+
+    {/* Login Actual Div */}
+    <div className='loginDiv'>
+    <form className='formDiv'>
+      <h3>Doctor Verse</h3>
+      <div className='form-element'>
+        <label>E-mail</label>
+        <input name='email' className='input' type='text' autoComplete='off' onChange={e => setDetails({...details, email: e.target.value})} value={details.email} required style={{color:'white'}}></input>
```
Comment (jain-rishabh-21) [inline client/src/pages/Login/index.js:48]: indent these lines properly

File: client/src/pages/Login/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      </div>
+      <div className='form-element'>
+        <label>Password</label>
+        <input name='password' className='input' type='text' autoComplete='off' onChange={e => setDetails({...details, password: e.target.value})} value={details.password} required style={{color:'white'}}></input>
+      </div>
+      {/* Form Buttons */}
+      <div className='FormButtonsDiv'>
+        <div className='signUp'><button className='formButton' 
+        style={{
+          backgroundColor: 'white',
```
Comment (jain-rishabh-21) [inline client/src/pages/Login/index.js:58]: use rbg codes for color

File: client/src/pages/Login/index.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          color: 'black',
+          border: '1px solid black',
+          borderRadius: '5px',
+          width: '100px',
```
Comment (jain-rishabh-21) [inline client/src/pages/Login/index.js:62]: keeping height & width in pixels will keep the website responsive?

File: client/src/pages/SignUp/index.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      else{
+        alert('Invalid Credentials')
+      }
+    } catch (error) {
+      console.error('Error:', error)
+    }
+  }
+
+  return (
+    <>
```
Comment (jain-rishabh-21) [inline client/src/pages/SignUp/index.js:39]: What is this?

File: client/src/pages/SignUp/index.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      <div className='form-element'>
+        <label>Create Password</label>
+        <input name='password' className='input' type='text' autoComplete='off' onChange={e => setDetails({...details, password: e.target.value})} value={details.password} style={{color:'white'}} required></input>
+      </div>
+      <div className='form-element'>
+        <label>Confirm Password</label>
+        <input name='password' className='input' type='text' autoComplete='off' onChange={e => setDetails({...details, confirmPassword: e.target.value})} value={details.confirmPassword} style={{color:'white'}} required></input>
+      </div>
+
+      {/* Form Buttons */}
```
Comment (jain-rishabh-21) [inline client/src/pages/SignUp/index.js:66]: Indent the whole file properly

File: client/src/pages/SignUp/login.css
Comment (jain-rishabh-21) [inline client/src/pages/SignUp/login.css:1]: Remove any unnecessary tag

**Review by jain-rishabh-21 — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 1 above)
Comment (Deepanshu0703) [reply client/src/App.js]: resolved.

File: client/src/pages/Login/index.js (hunk 2 above)
Comment (Deepanshu0703) [reply client/src/pages/Login/index.js]: resolved

File: client/src/pages/Login/index.js (hunk 3 above)
Comment (Deepanshu0703) [reply client/src/pages/Login/index.js]: done

File: client/src/pages/Login/index.js (hunk 4 above)
Comment (Deepanshu0703) [reply client/src/pages/Login/index.js]: changed in rem

File: client/src/pages/SignUp/index.js (hunk 6 above)
Comment (Deepanshu0703) [reply client/src/pages/SignUp/index.js]: done

File: client/src/pages/SignUp/login.css
Comment (Deepanshu0703) [reply client/src/pages/SignUp/login.css:1]: done

**Comment by jain-rishabh-21** [comment]
Can you resolve the conflicts? @Deepanshu0703

**Comment by Deepanshu0703** [comment]
Resolved the conflicts @jain-rishabh-21

**Comment by Deepanshu0703** [comment]
@jain-rishabh-21 Please look into it

**Comment by jain-rishabh-21** [comment]
Can you correct the PR title as well? See merged PRs for the title

**Comment by Deepanshu0703** [comment]
I have tried to change the name according to already merged PR but still this problem is occuring. Also I don't think PR title will cause any problem after merging.

**Comment by Deepanshu0703** [comment]
please check the changes @jain-rishabh-21

**Comment by Deepanshu0703** [comment]
Hey @jain-rishabh-21  Done

**Comment by Deepanshu0703** [comment]
Hey @jain-rishabh-21 i did all the changes please now merge it

**Comment by jain-rishabh-21** [comment]
Hey @Deepanshu0703 ! I still see 3 conflicts!

**Comment by jain-rishabh-21** [comment]
Can you ping me on linkedin as you do the rebase! I will merge it then and there only

**Comment by Deepanshu0703** [comment]
I have resolved these 3 conflicts too, can you plz send me your linkedin id so that i can connect if there is anything else i have to ask .
image

**Review by jain-rishabh-21 — APPROVED** [review]

**Comment by jain-rishabh-21** [comment]
I merged it! Thanks for the contribution!

**Pull request merged**